sw/device/silicon_creator/mask_rom/rsa_verify_unittest.cc

```diff
@@ -12,10 +12,307 @@
+                0xcedba143, 0xcf27b7e9, 0x432c2e73, 0x57138268, 0x9771655d,
+                0xdfd5054d, 0x80a69e65, 0xd8ca5b11, 0x64222c7f, 0x709e703b,
+                0x0452dae6, 0x2604c1bf, 0xaf29f6b5, 0x2773bf22, 0x83ab42d4,
+                0x34da57f5, 0xfad6aafc, 0xa23f2798, 0x88ab0542, 0x65219ceb,
+                0xc5fc703c, 0x9bab047a, 0x48749a33, 0x7067f6d5, 0xfcab7cc9,
+                0x878567df, 0x34e07abb, 0x6e5f5247, 0xdd57ed01, 0xd2cdc06e,
+                0x0b3c509c, 0x1c94f373, 0xc07a0024, 0x8c92383b, 0x575b4a5c,
+                0xd4c086fc, 0x27f19cd7, 0x496d70a4, 0x91d4b3cc, 0x73e34ca2,
+                0xa98f4fd4, 0x02ef38ac, 0xfb0a0675, 0xba14f83d, 0x0217c95b,
+                0xfc62ca77, 0x310b598c, 0x188e68cd, 0xdbcfdf58, 0xc783c009,
+                0xd8abae8c, 0x52d5f747, 0xee2dbda8, 0xd1f5ea87, 0x097f0e5b,
+                0x58407a2a, 0xfa880b9b, 0x528d2962, 0xa805f356, 0x9646688e,
+                0x2525612b, 0x900cacf4, 0xf844b2a4, 0x04007862, 0x96535db6,
+                0x25d03e7f, 0x4460bedf, 0x2961c014, 0x7a25057c, 0xf7bf0721,
+                0xfed9dbff, 0x7dfee1e2, 0xa6c7bcd3, 0x2cef3ab5, 0x7c7ffdf8,
+                0x4ab94057, 0x04c3cf7c, 0xf1022b35, 0x6cd62eae, 0x9e41a3b6,
+                0x8a31357b, 0x40013d2d, 0x5005f7c7, 0xa3ce1d53, 0xfe99692c,
+                0x8a612703, 0x2734ccde, 0xd115a702, 0x9b6c042c, 0xdd783f38,
+                0x5713d609,
+            },
+    },
+};
+
+class RsquareTest : public testing::TestWithParam<Rsquare> {};
+
+/**
+ * Tests that `kRsquares` is not missing any keys.
+ *
+ * Comparing the result with the expected value for each key provides
+ * practically full coverage for `calc_r_square` since it is not called with any
+ * other inputs.
+ */
+TEST(Rsquares, AllKeys) {
+  std::unordered_set<uint32_t> ids;
+  for (auto const &test_case : kRsquares) {
+    ids.insert(sigverify_rsa_key_id_get(test_case.key));
+  }
+
+  EXPECT_EQ(ids.size(), kSigVerifyNumRsaKeys);
+}
+
+TEST_P(RsquareTest, CalcRsquare) {
+  // Uninitialized on purpose.
+  sigverify_rsa_buffer_t res;
+  calc_r_square(GetParam().key, &res);
+  EXPECT_THAT(res.data, ::testing::ElementsAreArray(GetParam().r_square.data));
+}
+
+TEST_P(RsquareTest, MontMul) {
+  constexpr sigverify_rsa_buffer_t one{1};
+  // Uninitialized on purpose.
+  sigverify_rsa_buffer_t res1;
+  sigverify_rsa_buffer_t res2;
+
+  // mont_mul(R^2, 1, n) = R mod n
+  mont_mul(GetParam().key, &GetParam().r_square, &one, &res1);
+  // mont_mul(R, 1, n) = 1
+  mont_mul(GetParam().key, &res1, &one, &res2);
+
+  EXPECT_THAT(res2.data, ::testing::ElementsAreArray(one.data));
+}
+
+INSTANTIATE_TEST_SUITE_P(AllKeys, RsquareTest, testing::ValuesIn(kRsquares));
+
+/**
+ * PKCS #1 v1.5 encoded messages used in tests.
+ */
+
+/**
+ * Message: "test"
+ * SHA2-256 hash (little-endian): {0xb0f00a08, 0xd15d6c15, 0x2b0b822c,
+ * 0xa3bf4f1b, 0xc55ad015, 0x9a2feaa0, 0x884c7d65, 0x9f86d081}
+ */
```

Comment:
For this and the kSigTestCases below, add a short comment about how these were generated.  Its ok to refer people to your README doc instead of providing a detailed command here.  Something like:

"""
Generated by using the openssl command as discussed in the file .../README in section foo.
"""